src/Domain/LeanCode.DomainModels.EF/CachingEFRepository.cs

```diff
@@ -0,0 +1,67 @@
+using System.Diagnostics.CodeAnalysis;
+using LeanCode.DomainModels.Model;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Internal;
+
+namespace LeanCode.DomainModels.EF;
+
+public abstract class CachingEFRepository<TEntity, TIdentity, TContext> : EFRepository<TEntity, TIdentity, TContext>
+    where TEntity : class, IAggregateRootWithoutOptimisticConcurrency<TIdentity>
+    where TIdentity : notnull
+    where TContext : DbContext
+{
+    protected CachingEFRepository(TContext dbContext)
+        : base(dbContext) { }
+
+    /// <summary>
```

Comment:
Consider `<inheritdoc>`ing the docs from the `FindAsync` of the `EFRepository` and overridding necessary stuff.

Reply:
Ah yes, forgot about that. Will do!